lesson15/main.js

```diff
@@ -0,0 +1,97 @@
+"use strict";
+const wrapper = document.getElementById("js-wrapper");
+const ul = document.getElementById("js-ul");
+const submitBtn = document.getElementById("js-submitBtn");
+const modal = document.getElementById("js-modal");
+const openModalBtn = document.getElementById("js-openModalBtn");
+const jsonURL = "https://myjson.dit.upm.es/api/bins/j16z";
+const removeModal = () => {
+   modal.remove();
+   wrapper.classList.remove("bg_gray");
+};
+
+const startLoading = () => {
+   const loadingIcon = document.createElement("img");
+   loadingIcon.id = "loadingIcon";
+   loadingIcon.src = "img/loading-circle.gif";
+   wrapper.appendChild(loadingIcon);
+};
+
+const endLoading = () => {
+   const loadingIcon = document.getElementById("loadingIcon");
+   loadingIcon.remove();
+}
+
+openModalBtn.addEventListener("click", function () {
+   modal.style.display = "block";
+   wrapper.classList.add("bg_gray");
+   openModalBtn.remove();
+});
+
+const isEmptyString = (value) => value === "";
+
+submitBtn.addEventListener("click", () => {
+   const inputNumber = document.getElementById("js-number");
+   const inputText = document.getElementById("js-text");
+   if (isEmptyString(inputNumber.value) || isEmptyString(inputText.value)) { //inputNumber.valueとinputText.valueのデータ型がStringのため
+      alert("未入力です。");
+      return;
+   }
+   console.log(`ID:${inputNumber.value}`);
+   console.log(`ユーザー名:${inputText.value}`);
+   tryCreateElement();
+});
+
+async function init() {
+   removeModal();
+   startLoading();
+   try {
+      return await fetchData();
+   } catch (error) {
+      throw new Error(error);
+   } finally {
+      endLoading();
+   }
+}
+
+async function fetchData() {
+   const response = await fetch(jsonURL);
+   const json = await response.json();
+   return json.data;
+}
+
+const displayMessage = (message) => {
+   const div = document.createElement("div");
+   ul.appendChild(div);
+   div.textContent = message;
+}
+
+async function tryCreateElement() {
+   try {
+      const responseData = await init();
+      if (responseData.length === 0) {
+         const dataEmptyMessage = `データはまだありません。`;
+         displayMessage(dataEmptyMessage);
```

Comment:
nits 
I thought it would be better to just pass the string as an argument.
```suggestion
         displayMessage("データはまだありません");
```

Reply:
Thank you for your comment.
I thought that Sae-san was right.
422bc06

Do you have any criteria for making such a decision? If you do, please let me know for reference. Is it based on whether or not it would be incomprehensible to other people?

Reply:
displayMessageの引数名がmessgaeだったので、文字列のまま渡す方が わかりやすいと感じました。

また、変数にする理由もないと思いました。
例えば、判断しにくいもの(indexだったり、オブジェクトだったり)であれば、
変数に格納することで、名前を与えられるので 変数にする方がわかりやすいと感じます。

もっと理論的に説明できればいいのですが
感覚の部分が強くてすいません。

上記は答えではないので 参考程度にしてくれると嬉しいです…私も悩み悩みです😭

Reply:
丁寧に、ありがとうございます！今回は、さえさんの言う通り、そのまま渡すほうがわかりやすいとレビューを頂いてから気づきました。本当に何度もありがとうございます！